manifests/init.pp

```diff
@@ -421,6 +422,10 @@
     fail('ssh::sshd_config_banner must be set to be able to use sshd_banner_content.')
   }
 
+  if $ssh_gssapiauthentication != undef {
```

Comment:
Since the variable is set to `'yes'` by default instead of `undef` then the validation must occur. Please remove the if statement and leave in the validation. Without this, you could put in `undef` and it would pass and in the template you would have GSSAPIAuthentication without a value, which would cause problems for SSH.